FluentAvalonia/Styling/Core/FluentAvaloniaTheme.cs

```diff
@@ -383,16 +383,19 @@ private void Refresh(string newTheme)
         {
             if (newTheme == null)
             {
-                _requestedTheme = ResolveThemeAndInitializeSystemResources();                
+                newTheme = ResolveThemeAndInitializeSystemResources();
             }
 
             var old = _requestedTheme;
             if (!string.Equals(newTheme, old, StringComparison.OrdinalIgnoreCase))
             {
                 _requestedTheme = newTheme;
 
-                // Remove the old theme resources
-                _themeResources.MergedDictionaries.RemoveAt(1);
+                // Remove the old theme if any resources
+                if (_themeResources.Count > 0)
```

Comment:
There should always be 2 dictionaries within `_themeResources.MergedDictionaries`. In what case were you finding that not to be?

Reply:
It happens when you request a special Theme in App.axaml. I guess then the Theme wasn't loaded, so nothing to remove

In the demo app try 

```xml
<style:FluentAvaloniaTheme RequestedTheme="Light" />
```

Reply:
Ahh ok, yep. The root cause here is that setting `RequestedTheme` calls `Refresh()`, which can happen before everything initializes. We can keep the check you put in here since it doesn't hurt anything, but can you adjust the setter of `RequestedTheme` to:

```C#
set
{
    _requestedTheme = value;
    if (_hasLoaded)
       Refresh(value);
}
````

This will defer loading a new theme if we haven't initialized yet and will prevent that exception.

Reply:
good point 👍 

I've updated the PR now.